### ISSUE TITLE
Release first version of tensorflow-addons

### ISSUE THREAD

**Issue by seanpmorgan** [tensorflow/addons#56]
As we discussed in the February monthly meeting... we are targeting our first release by early next week. I wanted to get this issue up so we can discuss versioning and outline what else needs to get done.

## Versioning
I think there are two workable methods here:

1. Release on our own cadence and mention in the release which TF version it was tested against.
    - This is how it's done for [tensorflow_probability](https://github.com/tensorflow/probability/releases)
    - We could monitor our nightly tests and catch things like #37 which require us to rebuild
2. Release in step with core TensorFlow
   -  This is how it's done for [tensorboard](https://github.com/tensorflow/tensorboard/releases).
   - This is more intuitive for the end user IMO
   - Does require more of a commitment from the SIG

## Remaining blockers
- [X] Some examples (use tensorflow/examples as template)
- [X] Decide on version naming

**Comment by seanpmorgan** [comment]
cc @karmel @martinwicke

**Comment by armando-fandango** [comment]
In my opinion both approaches are fine but second approach of releasing in-step with TensorFlow is more preferable as it is more intuitive. I believe addons would be used more frequently with the core as compared to probability.

**Comment by facaiy** [comment]
I'm not quite sure what's the relation between addons and core:
1. addons depends on core?
2. core depends on addons?
3. they depend on each other?

**Comment by armando-fandango** [comment]
Number 1 : addons depends on core.

**Comment by martinwicke** [comment]
I think it's preferable to go with Option 1. Since we're only using public APIs (not something which has traditionally been true for TensorBoard) we shouldn't have to coordinate with TF releases (except for major releases). 

The less tied addons is to TF releases the better. The backwards compatibility guarantees in TF should make sure that addons doesn't break by a new TF release, so have a dependency on tensorflow>=X should be sufficient.

**Comment by lc0** [comment]
Option 1 sounds also better for me, as I would imagine `tf/addons`'s frequency of releases might be different than the one from TensorFlow. So better move faster and let people integrate their contributions without additional alignment.

You can still test against a couple of last versions of TF or new RCs

**Comment by dynamicwebpaige** [comment]
I'm also in support of Option 1. Addons releases should not be dependent on
the TensorFlow release cycle.

**Comment by facaiy** [comment]
I'm a little worried about the unclear overlap in scope between SIG/addons and `keras.*.experimental` API, as shown in PR https://github.com/tensorflow/addons/pull/14:
+ Will `keras.*.experimental` API depends on some latest features of addons in the future? cc @karmel

**Comment by martinwicke** [comment]
I think we should ideally limit keras.experimental to things that require
integration inside keras. This should be pretty rare. Or, things that we
know will end up in Keras proper withing one or maybe two releases. I think
this is the case here.

**Comment by karmel** [comment]
@facaiy -- we can't have Keras depend on Addons, but instead would move the overlapping features to one or the other, and direct users accordingly. In the case of #14 , we will remove the implementation from experimental and instead use Addons once we have unblocked some things internally.

Re the question at the top: one strategy we tried with TF models was to cut a release with each major TF release, and keep the numbers in sync, such that models head always worked with TF nightlies, and the releases worked with the same numbered TF. This meant we cut releases sometimes when there were no real changes in the models, but this did make tracking a little easier for users. One problem was that when we missed a release, or TF did a patch release, or we did a patch release, things became out of sync-- but in theory it would all normalize again with the next TF minor release. The bigger problem might be that TF releases are hard to time/predict, so it does require listening and waiting for changes to trigger a release on your side. Which is why we tend to encourage dependent libs like TF Probability to roll at their own pace.

**Comment by facaiy** [comment]
Thank you for your clarification, @martinwicke @karmel 

>  we can't have Keras depend on Addons

+1

> I think it's preferable to go with Option 1.

+1, I prefer option 1 as well. 

@av8ramit Hi, Amit, what do you think? By the way, the new process of tf-nightly has set up? And what's the release strategy of tf-2.0-preview?
+ https://github.com/tensorflow/community/pull/31
+ https://github.com/tensorflow/community/pull/30

**Comment by seanpmorgan** [comment]
Looks like we're leaning for option one so I just wanted to cc @csuter to see if you had any comments on how it's been working out for tf_probability

**Comment by csuter** [comment]
I think we are glad we are not strictly coupled to TF releases, although in practice we generally prepare a new release only when TF does. A crucial counterexample is that we will in the next few days be attempting to prepare a release that straddles compatibility with 1.13 and 2.0-preview. If we had tied ourselves strictly to TF (e.g., by numbering our versions in lockstep) this would be tricky. Happy to chat about our process and setup if anyone wants to block some time off. I can gladly share (internal) docs, too, if you haven't found them already.

**Comment by csuter** [comment]
Ah, I guess I incorrectly inferred that most on this thread were Googlers. It'll be difficult to meaningfully share the aforementioned internal docs externally.

**Comment by seanpmorgan** [comment]
No worries. We have a SIG meeting at 4pm EST this Friday if you can make it to that we could probably get you in and out in 10-15min if that helps. Otherwise we can just echo your sentiments.

**Comment by csuter** [comment]
Sure, send me an invite? (my google email is on my profile)

**Comment by seanpmorgan** [comment]
@dynamicwebpaige Can you mark this as Done in the TF2.0 tracker :D
